anomalydetection/detectors.py

```diff
@@ -0,0 +1,105 @@
+import pandas as pd
+
+
+class BaseDetector:
+    def __init__(self):
+        pass
+
+    def fit(self, data: pd.Series):
```

Comment:
Should we consider following sklearn, having both x and y as input, with y being optional? or is the plan to limit this package to supporting only unsupervised anomaly detection methods? I guess that would also be fine, for supervised approaches there are already many tools.

Reply:
I am not sure yet. detect should probably also be predict then. Let's decide later if we need to mix with sklearn